### PR TITLE
🧼 Channels as soft dependencies

### PR DESCRIPTION
**Pull request by rowanseymour**
Captura de Pantalla 2021-05-21 a la(s) 12 19 17

### CODE REVIEW

File: temba/api/v2/tests.py [hunk 1]
```diff
@@ -1473,7 +1473,7 @@ def test_channels(self):
 
         # create deleted channel
         deleted = Channel.create(self.org, self.admin, None, "JC", name="Deleted", address="nyaruka", role="SR")
-        deleted.release()
+        deleted.release(self.admin)
```
Comment (rowanseymour) [inline temba/api/v2/tests.py:1476]: unfortunately a lot of this diff is just making `user` an arg to `Channel.release`

File: temba/channels/models.py [hunk 2]
```diff
@@ -893,21 +893,30 @@ def claim(self, org, user, phone):
...
+
+        super().release(user)
 
         channel_type = self.get_type()
 
-        # release any channels working on our behalf as well
-        for delegate_channel in Channel.objects.filter(parent=self, org=self.org):
-            delegate_channel.release()
+        # ask the channel type to deactivate - as this usually means calling out to external APIs it can fail
+        if settings.IS_PROD:
```
Comment (rowanseymour) [inline temba/channels/models.py:906]: We should get rid of `settings.IS_PROD` - I nearly did it in this PR but the diff was already out of control. It made sense when we were regularly dumping prod databases and running them locally but nobody has done that for a while.

File: temba/orgs/models.py [hunk 3]
```diff
@@ -1650,15 +1650,9 @@ def release(self, *, release_users=True, delete=False):
         self.released_on = timezone.now()
         self.save(update_fields=("is_active", "released_on", "modified_on"))
 
-        # clear all our channel dependencies on our flows
-        for flow in self.flows.all():
-            flow.channel_dependencies.clear()
```
Comment (rowanseymour) [inline temba/orgs/models.py:1655]: `Channel.release` now does this

File: temba/channels/views.py [hunk 4]
```diff
@@ -1113,46 +1112,28 @@ def post(self, request, *args, **kwargs):
...
+                response["Temba-Success"] = self.cancel_url
                 return response
 
-            except ValueError as e:
-                logger.error("Error removing a channel", exc_info=True)
-
-                messages.error(request, str(e))
-                return HttpResponseRedirect(reverse("channels.channel_read", args=[channel.uuid]))
-
-            except Exception:  # pragma: no cover
```
Comment (rowanseymour) [inline temba/channels/views.py:1151]: the only exception type we don't swallow is `TwilioRestException` so these except blocks aren't actually used

File: temba/channels/models.py (hunk 2 above)
Comment (nicpottier) [reply temba/channels/models.py:906]: Ya, totally agree. Always takes me like four tries to get a telegram channel connected because of it too.

**Review by norkans7 — APPROVED** [review]

**Review by nicpottier — APPROVED** [review]

**Pull request merged**